### PR TITLE
Check if there is new core dump under /var/core post test

### PR DESCRIPTION
**Pull request by xwjiang-ms**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### What is the motivation for this PR?
Need to fail test case if there is any core dump generated post a test

#### How did you do it?
Compare the number of files under /var/core before test and after tear down
If there are new files, fail this test case.

#### How did you verify/test it?
Run a test

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1544,6 +1544,22 @@ def collect_db_dump(request, duthosts):
...
+    else:
+        pre_existing_cores = duthost.shell('ls /var/core/ | wc -l')['stdout']
+    
+    yield
+    if "20191130" in duthost.os_version:
+        coredumps_count = duthost.shell('ls /var/core/ | grep -v python | wc -l')['stdout']
+    else:
+        coredumps_count = duthost.shell('ls /var/core/ | wc -l')['stdout']
+    if int(coredumps_count) > int(pre_existing_cores):
+        raise pytest.fail("Core dumps found. Expected: {} Found: {}. Test failed".format(pre_existing_cores,\
```
Comment (ZhaohuiS) [inline tests/conftest.py:1560]: Does it need "raise" here? pytest.fail will fail the case.

Comment (xwjiang-ms) [reply tests/conftest.py:1560]: Oh right, I forgot to delete it.

File: tests/conftest.py [hunk 2]
```diff
@@ -1544,6 +1544,22 @@ def collect_db_dump(request, duthosts):
...
+        pre_existing_cores = duthost.shell('ls /var/core/ | wc -l')['stdout']
+    
+    yield
+    if "20191130" in duthost.os_version:
+        coredumps_count = duthost.shell('ls /var/core/ | grep -v python | wc -l')['stdout']
+    else:
+        coredumps_count = duthost.shell('ls /var/core/ | wc -l')['stdout']
+    if int(coredumps_count) > int(pre_existing_cores):
+        pytest.fail("Core dumps found. Expected: {} Found: {}. Test failed".format(pre_existing_cores,\
+            coredumps_count))
```
Comment (ZhaohuiS) [inline tests/conftest.py:1561]: Is it better to print new file name here?  If it fails, we are able to know which one is new core dump file and the timestamp could tell us when DUT generates file.

File: tests/conftest.py [hunk 3]
```diff
@@ -1544,6 +1544,22 @@ def collect_db_dump(request, duthosts):
     yield
     collect_db_dump_on_duts(request, duthosts)
 
+@pytest.fixture(autouse=True)
```
Comment (wangxin) [inline tests/conftest.py:1547]: It would be better to use scope 'module'.

Comment (xwjiang-ms) [reply tests/conftest.py]: Yes, that would be better

File: tests/conftest.py (hunk 2 above)
Comment (xwjiang-ms) [reply tests/conftest.py:1561]: I think it's unnecessary to print new file name. We won't collect core dumps to docker, if we want to know details of core dumps, we need to ssh to DUT, so I think it's unnecessary to find new file and print its name if we can simply use "ls -l" to get it.

**Review by wangxin — APPROVED** [review]

**Comment by ZhaohuiS** [comment]
@wangxin Can we include it this in 202012 branch? There is core dump file generated for `caclmgrd `during test `test_cacl_scale_rules_ipv6`.

**Pull request merged**